### PR TITLE
[RM-21][RM-22] user profile page

### PR DESCRIPTION
**Pull request by nels-dev**
+ allow users to view and update profile data
+ image upload is achieved using local storage combined with firebase storage
+ chats to display partners' profile image
+ removed action bar
+ if a logged-in user is present, skip the Sign In page.
+ user can sign out from the profile page

[rooftown_demo_RM21.webm](https://user-images.githubusercontent.com/101234025/222021325-3a4ea769-2b86-4b70-a9f5-dc3c6b5cc5c7.webm)

### CODE REVIEW

File: app/src/main/res/layout/fragment_profile.xml [hunk 1]
```diff
@@ -1,15 +1,170 @@
...
         android:layout_height="wrap_content"
-        android:text="@string/profileFragmentTitle"
-        android:textSize="25sp"
-        android:layout_centerInParent="true"/>
+        android:orientation="vertical">
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:orientation="horizontal">
```
Comment (vinztsang) [inline app/src/main/res/layout/fragment_profile.xml:20]: Others look fine, but should we make the profile picture center aligned here?

File: app/src/main/res/layout/fragment_edit_profile.xml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            <ImageView
+                android:id="@+id/imgViewEditProfileImage"
+                android:layout_width="150dp"
+                android:layout_height="150dp"
+                android:layout_weight="0"
+                android:foreground="@android:drawable/ic_menu_edit"
+                android:foregroundGravity="bottom|right"
+                android:scaleType="centerCrop"
+                app:srcCompat="@drawable/ic_outline_account_circle_24" />
+        </LinearLayout>
```
Comment (vinztsang) [inline app/src/main/res/layout/fragment_edit_profile.xml:31]: Does it also need to be center aligned if the profile page changes to center aligned?

**Review by vinztsang — COMMENTED** [review]
Others look good to me

File: app/src/main/res/layout/fragment_profile.xml (hunk 1 above)
Comment (nels-dev) [reply app/src/main/res/layout/fragment_profile.xml]: changed

File: app/src/main/res/layout/fragment_edit_profile.xml (hunk 2 above)
Comment (nels-dev) [reply app/src/main/res/layout/fragment_edit_profile.xml:32]: changed to center align

**Review by vinztsang — APPROVED** [review]
LGTM

**Pull request merged**